api/src/main/java/org/openmrs/module/mirebalais/MirebalaisHospitalActivator.java

```diff
@@ -15,221 +15,87 @@
+        this.daemonToken = daemonToken;
     }
 
     /**
      * @see ModuleActivator#started()
      */
+    @Override
     public void started() {
         try {
-
-            Config config = Context.getRegisteredComponents(Config.class).get(0); // currently only one of these
-
-            AdministrationService administrationService = Context.getAdministrationService();
-            ReportService reportService = Context.getService(ReportService.class);
-            ReportDefinitionService reportDefinitionService = Context.getService(ReportDefinitionService.class);
-            SerializedObjectDAO serializedObjectDAO = Context.getRegisteredComponents(SerializedObjectDAO.class).get(0);
-            PrinterService printerService = Context.getService(PrinterService.class);
-            DispositionService dispositionService = Context.getService(DispositionService.class);
-
-            removeOldPrivileges();
-
-            setDispositionConfig(config, dispositionService);
-
-            // register our custom print handlers
-            PrinterSetup.registerPrintHandlers(printerService);
-
-            // configure default dashboard in coreapps
-            updateGlobalProperty(CoreAppsConstants.GP_DASHBOARD_URL, config.getDashboardUrl());
-            // configure default visits page in coreapps
-            updateGlobalProperty(CoreAppsConstants.GP_VISITS_PAGE_URL, config.getVisitPageUrl());
-            // configure default specific visit detail page in coreapps
-            updateGlobalProperty(CoreAppsConstants.GP_VISITS_PAGE_WITH_SPECIFIC_URL, config.getVisitsPageWithSpecificUrl());
-
-            if (config.isComponentEnabled(Components.LEGACY_MPI)) {
-                LegacyMasterPatientIndexSetup.setupConnectionToMasterPatientIndex(customProperties);
-            }
-
-            if (config.isComponentEnabled(Components.ARCHIVES)) {
-                ArchivesSetup.setupCloseStaleCreateRequestsTask();
-                ArchivesSetup.setupCloseStalePullRequestsTask();
-            }
-
-            if (config.isComponentEnabled(Components.APPOINTMENT_SCHEDULING)) {
-                AppointmentSchedulingSetup.setupMarkAppointmentAsMissedOrCompletedTask();
-                if (config.getCountry().equals(ConfigDescriptor.Country.HAITI)) {
-                    AppointmentSchedulingSetup.customizeDailyAppointmentsDataSet();
-                }
-            }
-
-
-            if (config.isComponentEnabled(Components.RADIOLOGY) && config.getSite().equalsIgnoreCase("MIREBALAIS")) {
-                updateGlobalProperty(OpenmrsConstants.GP_ORDER_NUMBER_GENERATOR_BEAN_ID, MirebalaisConstants.RADIOLOGY_ORDER_NUMBER_GENERATOR_BEAN_ID);
+            log.info("Mirebalais Module Started, initiating configuration");
+
+            final ConfigurationSetup configurationSetup = Context.getRegisteredComponents(ConfigurationSetup.class).get(0);
+            configurationSetup.setupBase();
+
+            if (runInSeparateThread()) {
```

Comment:
why are we running all of this in a separate thread?  the idea was just to run the "slow" loading stuff, like Concepts, in a separate thread?  Have you tested firing up a system from scratch with "separate thread" enabled?

Reply:
No, I have not tested a new system from scratch yet.  But this "separate thread" thing was always an incredible hack that could lead to lots of unpredictable system instability and I think we should be trying to eliminate it altogether.  I guess it would be easy to keep the existing behavior - just need to separate things out in the configuration setup class.  I'll do that so that we can get past this question of compatibility and impact, but I think we should seriously look at this approach in general - soon.

Reply:
Yeah, as the separate thread was a just a hack to avoid having extended downtime at Mirebalais due to concept loading, let's keep it targeted towards those few things.  Theoretically, when we switch to OCL we will no longer need this.